include/Gaffer/Context.inl

```diff
@@ -174,24 +172,21 @@ inline bool Context::internalSet( const IECore::InternedString &name, const Valu
 		// expense of checking.
 		m_map[name] = value;
 		m_hashValid = false;
-		return true;
 	}
 	else
 	{
-		// Avoid emitting `changedSignal` if the value hasn't
-		// actually changed. We want to avoid expensive re-evaluations
-		// that might otherwise be triggered in the UI.
+		// Always assign to the value, because the caller might be updating
+		// `m_allocMap` to match.
```

Comment:
It feels extremely nitpicky to be checking verb tenses, but it maybe does help to be maximally clear here: would be it more accurate to say `because the caller might have updated 'm_allocMap' and removed the previous value`

Reply:
I think the extra bit of clarity that brings is worthwhile - thanks for suggesting it. Squashed into https://github.com/GafferHQ/gaffer/pull/5367/commits/520c7a35d8818dba0a9f37bd36fbf8ce3dd2678d.